src/packages/@ncigdc/utils/getUsefulFacets.tsx

```diff
@@ -0,0 +1,38 @@
+import { omitBy, some } from 'lodash';
+
+type TFacet =
+  | Record<string, ICategoricalFacet>
+  | Record<string, IContinuousFacet>;
+
+interface IContinuousFacet {
+  stats: {
+    count: number | null;
+    max: number | null;
+    min: number | null;
+    avg: number | null;
+    sum: number | null;
+  };
+  count?: number;
+}
+
+interface ICategoricalFacet {
+  buckets: [{ key: string; doc_count: number }];
+}
+
+export default (facets: TFacet) =>
+  omitBy(
+    facets,
+    aggregation =>
```

Comment:
How about assign type to the aggregation here(like (aggregation: ICategoricalFacet)=> ...) instead of assign type to everyone in the function?

Reply:
Unfortunately because there's 2 possible structures for aggregations (continuous vs categorical), i need to cast the type each time a property is accessed